File: solutions/clojure/bob/2/src/bob.clj
(ns bob
  (:require [clojure.string :as str]))

(defn response-for [s]
  (let [s (str/trim s)
        letter? #(Character/isLetter %)
        upper-case? #(Character/isUpperCase %)
        letters (filter letter? s)
        yell? (and 
               (seq letters) 
               (every? upper-case? letters))]
    (cond
      (and yell? (str/ends-with? s "?")) "Calm down, I know what I'm doing!"
      yell? "Whoa, chill out!"
      (str/ends-with? s "?") "Sure."
      (str/blank? s) "Fine. Be that way!"
      :else "Whatever.")))

File: solutions/cpp/bob/2/bob.cpp
#include "bob.h"


namespace bob {

    std::string hey(std::string stringIn) {
        bool question = false;
        bool yelled = true;
        bool silence = true;

        for (char c : stringIn) {
            if (std::islower(c)) {
                yelled = false;
            }

            if (c == '?') {
                question = true;
            }

            if (std::isspace(c)) {
                question = false;
            } else {
                silence = false;
            }
        }


        if (question && yelled) {
            return "Calm down, I know what I'm doing!";
        } else if (silence) {
            return "Fine. Be that way!";
        } else if (question) {
            return "Sure.";
        } else if (yelled) {
            return "Whoa, chill out!";
        }
        return "Whatever."; 
    }

}  // namespace bob
